aspnetcore/blazor/components/render-modes.md

```diff
@@ -0,0 +1,386 @@
+* You can't switch to a different interactive render mode in a child component. For example, a Server component can't be a child of a WebAssembly component.
+* Parameters passed to an interactive child component from a Static parent must be JSON serializable. This means that you can't pass render fragments or child content from a Static parent component to an interactive child component.
+
+### Render mode inheritance
+
+Consider the following non-routable, non-page `SharedMessage` component for use in other components. The render mode agnostic `SharedMessage` component doesn't apply a render mode with an [`@attribute` directive](xref:mvc/views/razor#attribute).
+
+`SharedMessage.razor`:
+
+```razor
+<button @onclick="UpdateMessage">Click me</button> @message
+
+@code {
+    private string message = "Not clicked yet.";
+
+    private void UpdateMessage()
+    {
+        message = "Somebody clicked me!";
+    }
+}
+```
+
+If the `SharedMessage` component is placed in a statically-rendered parent component, the `SharedMessage` component is also rendered statically and isn't interactive. The button doesn't call `UpdateMessage`, and the message isn't updated.
+
+`PassedRenderMode1.razor`:
+
+```razor
+@page "/passed-render-mode-1"
+
+<SharedMessage />
+```
+
+If the `SharedMessage` component is placed in a component that defines the render mode, it inherits the applied render mode.
+
+In the following example, the `SharedMessage` component is interactive over a SignalR connection to the client. The button calls `UpdateMessage`, and the message is updated.
+
+`PassedRenderMode2.razor`:
+
+```razor
+@page "/passed-render-mode-2"
+@attribute [RenderModeServer]
+
+<SharedMessage />
+```
+
+### A parent component with two child sibling component that use different render modes
+
+
+
+### A negative case of a child component that tries to use a different interactive render mode than its parent, which results in an error
+
+
+
+### A parent component with an interactive child component that takes a parameter that is serializable (works)
+
+
+
+### A negative case of an interactive child component that has a non-serializable component parameter, like child content or a render fragment, which results in an error.
+
+-----> Show that you can sometimes work around this by wrapping the child component in another component that doesn't have the parameter. This is what we do in the Blazor Web App template with the Routes component to wrap the Blazor Router.
+
+
+
+## Set the render mode for the entire app
+
+<!-- UPDATE 8.0 RC2: Remove preview remark and update code -->
+
+To set the render mode for the entire app, indicate the render mode at the highest level component in the app's component hierarchy, typically the `Routes` component (`Components/App.razor`) for apps based on the Blazor Web App project template:
```

Comment:
```suggestion
To set the render mode for the entire app, indicate the render mode at the highest-level component in the app's component hierarchy that isn't a root component (root components cannot be interactive). Typically, this will be the `Routes` component (`Components/App.razor`) for apps based on the Blazor Web App project template:
```